[Contextual Security] replace node label and detail labels tooltips with popovers

alexreal1314 opened the pull request:
This PR closes the following tickets:
[Migrate tooltips to popovers in label nodes](https://github.com/elastic/kibana/issues/233881)
[Show IP flyout preview when clicking on any IP within entity/label IPs popover](https://github.com/elastic/kibana/issues/237678)


## Summary
This pull request enhances the graph investigation feature by introducing new popover components for displaying additional node details such as IPs, country flags, and event/alert analysis. These changes improve the user experience by allowing users to interactively view more granular information about graph nodes. The implementation includes new hooks, handlers, and integration with existing node types.

**Popover Feature Additions**

* Added new popover hooks and components: `useIpPopover`, `useCountryFlagsPopover` which both use `useNodeDetailsPopover` hook, and `useEventDetailsPopover`, enabling interactive popover displays for IP addresses, country flags, and event details in the graph investigation UI.
* Integrated popover handlers (`createIpClickHandler`, `createCountryClickHandler`, `createEventClickHandler`) into node rendering logic, allowing entity and label nodes to trigger corresponding popovers on click.

**Graph Investigation Component Enhancements**

* Updated the `GraphInvestigation` component to manage popover state for IPs, country codes, and event analysis, and to render the new popover components alongside existing ones.
* extended popover open/close logic to ensure only one popover is open at a time, now including the new popover types in the check.

**Storybook Improvements**

* Enhanced the `CountryFlags` Storybook stories to showcase the new popover functionality, including a new story (`CountryFlagsWithPopover`) that demonstrates interactive country flag popover.
* Enhanced the `Ips` Storybook stories to showcase the new popover functionality, including a new story (`IpsWithPopover`) that demonstrates interactive ip popover.
* Enhanced the `LabelNode -> badges` storybook

**Generic Popover Utility**

* Introduced a generic popover utility (`useNodeDetailsPopover`) to simplify the creation and management of popover components for displaying lists of node details.


### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be allowlisted in the cloud and added to the [docker list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking changes have been approved by the breaking-change committee. The `release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was used on any tests changed
- [x] The PR  description includes the appropriate Release Notes section, and the correct `release_note:*` label is applied per the [guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [x] Review the [backport guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing) and apply applicable `backport:*` labels.

### How to test

1. Run Elasticsearch with `yarn es snapshot --license trial -E reindex.remote.whitelist=keep-long-live-env-ess.es.us-west2.gcp.elastic-cloud.com:443`
2. Run Kibana locally with `yarn start --no-base-path`
3. Install Cloud Asset Discovery (skip agent installation)
4. Re-index Asset Inventory data with this query
    ```
    POST _reindex?wait_for_completion=true
    {
      "conflicts": "proceed", 
      "source": {
        "remote": {
          "host": "${ES_REMOTE_HOST}", # <-- Set var in Config tab to https://keep-long-live-env-ess.es.us-west2.gcp.elastic-cloud.com:443
          "username": "${ES_REMOTE_USER}", # <-- Set var in Config tab to your user in https://keep-long-live-env-ess.kb.us-west2.gcp.elastic-cloud.com/app/management/security/users i.e. alberto
          "password": "${ES_REMOTE_PASS}" # <-- Set var in Config tab to your password in the same cloud env
        },
        "index": "logs-cloud_asset_inventory*",
        "query": {
          "bool": {
            "must": [
              {
                "range": {
                  "@timestamp": {
                    "gte": "now-1d"
                  }
                }
              }
            ]
          }
        }
      },
      "dest": {
        "op_type": "create",
        "index": "logs-cloud_asset_inventory.asset_inventory-default"
      }
    }
    ```
5. Reindex AWS Cloudtrail logs with this query
    ```
    POST _reindex?wait_for_completion=false
    {
      "conflicts": "proceed", 
      "source": {
        "remote": {
          "host": "${ES_REMOTE_HOST}",
          "username": "${ES_REMOTE_USER}",
          "password": "${ES_REMOTE_PASS}"
        },
        "index": "logs-aws.cloudtrail-*",
        "query": {
           "bool": {
             "must": [],
             "filter": [
               {
                 "range": {
                   "@timestamp": {
                     "gte": "now-3h",
                     "lte": "now"
                   }
                 }
               }
             ]
           }
         }
      },
      "dest": {
        "op_type": "create",
        "index": "logs-aws.cloudtrail-reindexed"
      }
    }
    ```
6. Create a new Data View that matches this index pattern `logs-aws.cloudtrail-*`. Give it the exact same name
7. Open the data-view you created and copy the id from the url
<img width="1700" height="906" alt="image" src="https://github.com/user-attachments/assets/c3fbf289-9a86-491d-97bb-2b52daf80e13" />

8. Download and decompress this exported rule: [rules_export (2).ndjson.zip](https://github.com/user-attachments/files/22530457/rules_export.2.ndjson.zip). Replace the value of "data_view_id" with the id you copied in the previous step
<img width="659" height="430" alt="image" src="https://github.com/user-attachments/assets/9632b750-0499-49a0-aa9a-9af5fac93839" />

9. Go to Rules page and click on "Import rule"
10. Go to Alerts page. Set the date range from several days ago until now. You'll see a bunch of alerts.
11. Open one alert flyout and check you can expand the Graph Visualization tab.
12. Click on all options to show related events, related entities, show actions on this entity, etc.... try to show as many nodes as possible and check they're grouped.
<img width="1459" height="930" alt="image" src="https://github.com/user-attachments/assets/6667dea8-40a8-4da5-af4e-86f5b7ac9599" />

### Screenshots

country flags:

<img width="1654" height="931" alt="image" src="https://github.com/user-attachments/assets/b76e2fe5-3c1f-4c59-abc5-f250f13631ee" />

ips popover + network preview popover:

<img width="1648" height="933" alt="image" src="https://github.com/user-attachments/assets/3d538312-de09-4dce-a8e6-15e4ce313767" />

only events popover:

<img width="1454" height="933" alt="image" src="https://github.com/user-attachments/assets/0753f6e1-0d99-4872-b82f-7e9b595c3530" />

events and alerts storybook:

<img width="1445" height="935" alt="image" src="https://github.com/user-attachments/assets/3007bfc6-8d0e-4402-8dd7-b105340d565d" />

events and alerts popover: 

<img width="1472" height="934" alt="image" src="https://github.com/user-attachments/assets/d8470e3c-7571-4f30-b5de-75c5dd7d6a52" />

alexreal1314 committed:
add ip popover component and use the centralized popver management state in graph investigation component

update ip component unit tests and storybook

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

alexreal1314 committed:
create useNodeDetailsPopover hook for reusable popover functionality

update countryCodes component to pass on click handler to the popover

update related unit tests

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/347383)
* Commit: 4c58133c570afb4bca25275fbfd372f4e8ccc5be
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-236906)

### Failed CI Steps
* [Jest Tests #6](https://buildkite.com/elastic/kibana-pull-request/builds/347383#0199b331-edb7-455b-9a54-41978e32614c)
* [Jest Tests #6](https://buildkite.com/elastic/kibana-pull-request/builds/347383#0199b352-70dc-45a8-9f9f-285bf69b6535)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/347383#0199b331-edb7-455b-9a54-41978e32614c) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/347383/jobs/0199b331-edb7-455b-9a54-41978e32614c/artifacts/0199b352-5a9f-4858-ab16-1b632da72615) Jest Tests #<span></span>6 / defend_insights_loader getDefendInsightsDocsCount returns correct count
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/347383#0199b352-70dc-45a8-9f9f-285bf69b6535) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/347383/jobs/0199b352-70dc-45a8-9f9f-285bf69b6535/artifacts/0199b35a-bb0c-4079-8c1f-ab2a6725e6bf) Jest Tests #<span></span>6 / defend_insights_loader getDefendInsightsDocsCount returns correct count

### History
* :broken_heart: [Build #345693](https://buildkite.com/elastic/kibana-pull-request/builds/345693) failed b659ccc102a1108a3d911a13cdac12ae989bf0b5

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/347383","number":347383,"commit":"4c58133c570afb4bca25275fbfd372f4e8ccc5be"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/345693","number":345693,"commit":"b659ccc102a1108a3d911a13cdac12ae989bf0b5"}],"number":347383}
buildkite-pr-comment-->

alexreal1314 committed:
replace eui links with empty buttons in ip and country codes components

alexreal1314 committed:
Replace tooltip with popover when clicking on event badge in label nodes

update related storybook and unit tests

alexreal1314 committed:
add click event callback to AlertCountBadge

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/347410)
* Commit: 02e54c475312c71cf4e5d818bbd42fec7a031d05

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/347410#0199b61f-b858-460b-969d-98617225c820)

### History
* :broken_heart: [Build #347396](https://buildkite.com/elastic/kibana-pull-request/builds/347396) failed 953cb3e8b7ddc6b2e160a9a37f4849531832a09f
* :broken_heart: [Build #347383](https://buildkite.com/elastic/kibana-pull-request/builds/347383) failed 4c58133c570afb4bca25275fbfd372f4e8ccc5be
* :broken_heart: [Build #345693](https://buildkite.com/elastic/kibana-pull-request/builds/345693) failed b659ccc102a1108a3d911a13cdac12ae989bf0b5

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/347410","number":347410,"commit":"02e54c475312c71cf4e5d818bbd42fec7a031d05"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/347396","number":347396,"commit":"953cb3e8b7ddc6b2e160a9a37f4849531832a09f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/347383","number":347383,"commit":"4c58133c570afb4bca25275fbfd372f4e8ccc5be"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/345693","number":345693,"commit":"b659ccc102a1108a3d911a13cdac12ae989bf0b5"}],"number":347410}
buildkite-pr-comment-->

alexreal1314 committed:
update node details popover logic

alexreal1314 committed:
update label node badges storybook with different event/alert cases

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/347412)
* Commit: cfb40b85678b9e57429a3992870e60b30fe98ee9

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/347412#0199b63d-11cc-4692-add5-7f8e84c87b92)

### History
* :broken_heart: [Build #347410](https://buildkite.com/elastic/kibana-pull-request/builds/347410) failed 02e54c475312c71cf4e5d818bbd42fec7a031d05
* :broken_heart: [Build #347396](https://buildkite.com/elastic/kibana-pull-request/builds/347396) failed 953cb3e8b7ddc6b2e160a9a37f4849531832a09f
* :broken_heart: [Build #347383](https://buildkite.com/elastic/kibana-pull-request/builds/347383) failed 4c58133c570afb4bca25275fbfd372f4e8ccc5be
* :broken_heart: [Build #345693](https://buildkite.com/elastic/kibana-pull-request/builds/345693) failed b659ccc102a1108a3d911a13cdac12ae989bf0b5

cc @alexreal1314

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/347412","number":347412,"commit":"cfb40b85678b9e57429a3992870e60b30fe98ee9"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/347410","number":347410,"commit":"02e54c475312c71cf4e5d818bbd42fec7a031d05"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/347396","number":347396,"commit":"953cb3e8b7ddc6b2e160a9a37f4849531832a09f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/347383","number":347383,"commit":"4c58133c570afb4bca25275fbfd372f4e8ccc5be"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/345693","number":345693,"commit":"b659ccc102a1108a3d911a13cdac12ae989bf0b5"}],"number":347412}
buildkite-pr-comment-->

alexreal1314 commented on x-pack/solutions/security/packages/kbn-cloud-security-posture/graph/src/components/node/label_node/label_node_badges.tsx:
```diff
@@ -46,18 +64,45 @@ const AlertIcon: React.FC<{ color: string }> = ({ color }) => (
 );
 
-const EventBadge: React.FC<{ count: number }> = ({ count }) => {
+const EventBadge: React.FC<{
```
currently same popover is used for events and alerts - see [comment](https://github.com/elastic/kibana/issues/233881#issuecomment-3369242110) for reasons specified in the comment. we can separate to two separate popovers - one for events data and another for alerts data depending on the decision.

alexreal1314 commented on x-pack/solutions/security/packages/kbn-cloud-security-posture/graph/src/components/node/label_node/label_node_badges.stories.tsx:
```diff
@@ -28,21 +30,50 @@ const useCases = {
   'Multiple events and alerts': { uniqueEventsCount: 2, uniqueAlertsCount: 2 },
   'Hundreds of events and alerts': { uniqueEventsCount: 120, uniqueAlertsCount: 120 },
   'Millions of events and alerts': { uniqueEventsCount: 1_200_000, uniqueAlertsCount: 1_200_000 },
+  'With clickable events popover - only events': { uniqueEventsCount: 5, uniqueAlertsCount: 0 },
```
will update storybook if necessary according to decision - see `[comment](https://github.com/elastic/kibana/pull/236906/files#r2404681041)`.

kibanamachine committed:
[CI] Auto-commit changed files from 'security: 3rd-party dependencies'

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/347415)
* Commit: 479624e0253e70e1fb8e2508b3643049a27c3496

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/347415#0199b660-0fc5-4f1f-bd81-a064dafdb086)

### History
* :broken_heart: [Build #347412](https://buildkite.com/elastic/kibana-pull-request/builds/347412) failed cfb40b85678b9e57429a3992870e60b30fe98ee9
* :broken_heart: [Build #347410](https://buildkite.com/elastic/kibana-pull-request/builds/347410) failed 02e54c475312c71cf4e5d818bbd42fec7a031d05
* :broken_heart: [Build #347396](https://buildkite.com/elastic/kibana-pull-request/builds/347396) failed 953cb3e8b7ddc6b2e160a9a37f4849531832a09f
* :broken_heart: [Build #347383](https://buildkite.com/elastic/kibana-pull-request/builds/347383) failed 4c58133c570afb4bca25275fbfd372f4e8ccc5be
* :broken_heart: [Build #345693](https://buildkite.com/elastic/kibana-pull-request/builds/345693) failed b659ccc102a1108a3d911a13cdac12ae989bf0b5

cc @alexreal1314

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/347415","number":347415,"commit":"479624e0253e70e1fb8e2508b3643049a27c3496"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/347412","number":347412,"commit":"cfb40b85678b9e57429a3992870e60b30fe98ee9"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/347410","number":347410,"commit":"02e54c475312c71cf4e5d818bbd42fec7a031d05"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/347396","number":347396,"commit":"953cb3e8b7ddc6b2e160a9a37f4849531832a09f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/347383","number":347383,"commit":"4c58133c570afb4bca25275fbfd372f4e8ccc5be"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/345693","number":345693,"commit":"b659ccc102a1108a3d911a13cdac12ae989bf0b5"}],"number":347415}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/347416)
* Commit: 9da186a565396a0d0d0789b7b4f4f955fb778e80

### Failed CI Steps
* [Linting](https://buildkite.com/elastic/kibana-pull-request/builds/347416#0199b66f-4075-47b9-beaa-ae43e0d5e392)

### History
* :broken_heart: [Build #347415](https://buildkite.com/elastic/kibana-pull-request/builds/347415) failed 479624e0253e70e1fb8e2508b3643049a27c3496
* :broken_heart: [Build #347412](https://buildkite.com/elastic/kibana-pull-request/builds/347412) failed cfb40b85678b9e57429a3992870e60b30fe98ee9
* :broken_heart: [Build #347410](https://buildkite.com/elastic/kibana-pull-request/builds/347410) failed 02e54c475312c71cf4e5d818bbd42fec7a031d05
* :broken_heart: [Build #347396](https://buildkite.com/elastic/kibana-pull-request/builds/347396) failed 953cb3e8b7ddc6b2e160a9a37f4849531832a09f
* :broken_heart: [Build #347383](https://buildkite.com/elastic/kibana-pull-request/builds/347383) failed 4c58133c570afb4bca25275fbfd372f4e8ccc5be
* :broken_heart: [Build #345693](https://buildkite.com/elastic/kibana-pull-request/builds/345693) failed b659ccc102a1108a3d911a13cdac12ae989bf0b5

cc @alexreal1314

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/347416","number":347416,"commit":"9da186a565396a0d0d0789b7b4f4f955fb778e80"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/347415","number":347415,"commit":"479624e0253e70e1fb8e2508b3643049a27c3496"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/347412","number":347412,"commit":"cfb40b85678b9e57429a3992870e60b30fe98ee9"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/347410","number":347410,"commit":"02e54c475312c71cf4e5d818bbd42fec7a031d05"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/347396","number":347396,"commit":"953cb3e8b7ddc6b2e160a9a37f4849531832a09f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/347383","number":347383,"commit":"4c58133c570afb4bca25275fbfd372f4e8ccc5be"}],"number":347416}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/347472)
* Commit: c7108f8193174a3ad557e38d50ba888ea001dfc8

### Failed CI Steps
* [Investigations - Security Solution Cypress Tests #8](https://buildkite.com/elastic/kibana-pull-request/builds/347472#0199b884-a0d9-4158-8739-80bcbb6201cc)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/347472#0199b884-a0d9-4158-8739-80bcbb6201cc) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/347472/jobs/0199b884-a0d9-4158-8739-80bcbb6201cc/artifacts/0199b8bc-3efb-458e-9afb-9bea2253ac15) Investigations - Security Solution Cypress Tests #<span></span>8 / Timeline should verify add to timeline and investigate in timeline work from various places should verify add to timeline and investigate in timeline work from various places

### History
* :broken_heart: [Build #347416](https://buildkite.com/elastic/kibana-pull-request/builds/347416) failed 9da186a565396a0d0d0789b7b4f4f955fb778e80
* :broken_heart: [Build #347415](https://buildkite.com/elastic/kibana-pull-request/builds/347415) failed 479624e0253e70e1fb8e2508b3643049a27c3496
* :broken_heart: [Build #347412](https://buildkite.com/elastic/kibana-pull-request/builds/347412) failed cfb40b85678b9e57429a3992870e60b30fe98ee9
* :broken_heart: [Build #347410](https://buildkite.com/elastic/kibana-pull-request/builds/347410) failed 02e54c475312c71cf4e5d818bbd42fec7a031d05
* :broken_heart: [Build #347396](https://buildkite.com/elastic/kibana-pull-request/builds/347396) failed 953cb3e8b7ddc6b2e160a9a37f4849531832a09f

cc @alexreal1314

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/347472","number":347472,"commit":"c7108f8193174a3ad557e38d50ba888ea001dfc8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/347416","number":347416,"commit":"9da186a565396a0d0d0789b7b4f4f955fb778e80"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/347415","number":347415,"commit":"479624e0253e70e1fb8e2508b3643049a27c3496"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/347412","number":347412,"commit":"cfb40b85678b9e57429a3992870e60b30fe98ee9"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/347410","number":347410,"commit":"02e54c475312c71cf4e5d818bbd42fec7a031d05"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/347396","number":347396,"commit":"953cb3e8b7ddc6b2e160a9a37f4849531832a09f"}],"number":347472}
buildkite-pr-comment-->

alexreal1314 committed:
expand useIpPopover to allow opening network preview popover

kibanamachine committed:
[CI] Auto-commit changed files from 'security: 3rd-party dependencies'

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint_all_files --no-cache --fix'

alexreal1314 committed:
remove code duplication in country flags component and remove unused const

alexreal1314 commented on x-pack/solutions/security/packages/kbn-cloud-security-posture/graph/src/components/node/country_flags/country_flags.test.tsx:
@albertoblaz The reason I removed those tests it because of two reasons:
1. since the popover is somewhat external to the country flags component implementation as we just inject data into it I don't think this functionality needs to be tested here. The rendering of the popover should be tested separately IMO - ill add some testing in there.
2. I see that all testing related to conversion of country codes with upper/lower/mixed casing and localization is tested in the [country_codes.test.ts](https://github.com/elastic/kibana/blob/main/x-pack/solutions/security/packages/kbn-cloud-security-posture/graph/src/components/node/country_flags/country_codes.test.ts) file?

WDYT?

albertoblaz reviewed (commented):
I'm thinking we should also check the Network preview panel opens when an IP is clicked. That way, we know for sure the whole interaction is working. 

This would require:
1. Adding some fake IPs to `source.ip` or `host.ip` fields in the `security_alerts_modified_mappings` fixtures
2. Extend these e2e tests in the `x-pack/solutions/security/test/cloud_security_posture_functional/pages/` folder asserting the panel opens:
    - `events_flyout.ts`
    - `alerts_flyout.ts`

Let me know if you need any help to finish it while you're on vacation ok?

alexreal1314 commented on x-pack/solutions/security/packages/kbn-cloud-security-posture/graph/src/components/graph_investigation/use_node_details_popover.tsx:
```diff
@@ -0,0 +1,135 @@
+  itemTestSubj,
+  popoverTestSubj,
+}: UseGenericPopoverProps): UseNodeDetailsPopoverReturn => {
+  const { id, state, actions } = useGraphPopover(popoverId);
```
> Why do we pass a dynamic `popoverId` to `useGraphPopover` here? In `useEventDetailsPopover` we're passing a static `"events-popover"` id instead

this is because `useNodeDetailsPopover` is being reused by both ips country_flags components. And could be used by other types of components who represent details of a node so the id needs to be injected to create different popover contexts.
On the other hand `useEventDetailsPopover` is used to handle only the event popover. could be modified if needed if this hook is being reused but for now we just pass it internally.

alexreal1314 commented on x-pack/solutions/security/packages/kbn-cloud-security-posture/graph/src/components/node/ips/ips.test.tsx:
we are not testing here the popover content, I'll add a test for it separately.

alexreal1314 commented on x-pack/solutions/security/packages/kbn-cloud-security-posture/graph/src/components/node/ips/ips.tsx:
```diff
@@ -6,46 +6,71 @@
 
 export const VISIBLE_IPS_LIMIT = 1;
-export const MAX_IPS_IN_TOOLTIP = 10;
+export const MAX_IPS_IN_POPOVER = 10;
```
good catch.

alexreal1314 committed:
fix styles in node details popover and added unit tests

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint_all_files --no-cache --fix'

alexreal1314 committed:
fix ips popover rendering

alexreal1314 committed:
add graph ftr to test ips popover functionality

fix esql query builder to return ips and country when no enriched data exists

albertoblaz reviewed (commented):
Overall it looks pretty good but I would:
- Address a couple of visual issues
- Update the badges storybook to show how popovers should behave against counters
- Remove the ESQL query change from this whole change-set

alexreal1314 committed:
update label node badges storybook and fix pr comments

alexreal1314 committed:
fix linting issues

alexreal1314 committed:
add tooltip when label node is truncated

alexreal1314 commented on x-pack/solutions/security/packages/kbn-cloud-security-posture/graph/src/components/graph_investigation/use_event_details_popover.tsx:
```diff
@@ -0,0 +1,100 @@
+            `}
+            truncation="end"
+            text={text}
+          />
```
@albertoblaz discussed with @larissaoliveiraelastic we will show the tooltip when the label text is truncated as seen in the storybook, ive made the updates to support it:
<img width="409" height="146" alt="image" src="https://github.com/user-attachments/assets/3f922eda-d8ff-4b9d-b270-acf2f60df5dd" />

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/351180)
* Commit: 6bbf454cbde11b62e6979ecf1edacdda2097d2ce
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-236906)
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-236906-6bbf454cbde1`

### History
* :green_heart: [Build #349965](https://buildkite.com/elastic/kibana-pull-request/builds/349965) succeeded 3958fb252c56eb8516193341ee01b28726be8496
* :yellow_heart: [Build #349930](https://buildkite.com/elastic/kibana-pull-request/builds/349930) was flaky 5f34c7e85a49f8786aa7b3a845cbab3a6e6965fc
* :green_heart: [Build #349223](https://buildkite.com/elastic/kibana-pull-request/builds/349223) succeeded 319b801ec53c57c54d535a9a2390436a05244028
* :broken_heart: [Build #349077](https://buildkite.com/elastic/kibana-pull-request/builds/349077) failed 30a97209f32df0498b5cafb8037262d4022785d9
* :broken_heart: [Build #348983](https://buildkite.com/elastic/kibana-pull-request/builds/348983) failed 00d2e653a3c2e409748592ab01742e42ac2d5632

cc @alexreal1314

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/351180","number":351180,"commit":"6bbf454cbde11b62e6979ecf1edacdda2097d2ce"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/349965","number":349965,"commit":"3958fb252c56eb8516193341ee01b28726be8496"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/349930","number":349930,"commit":"5f34c7e85a49f8786aa7b3a845cbab3a6e6965fc"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/349223","number":349223,"commit":"319b801ec53c57c54d535a9a2390436a05244028"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/349077","number":349077,"commit":"30a97209f32df0498b5cafb8037262d4022785d9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/348983","number":348983,"commit":"00d2e653a3c2e409748592ab01742e42ac2d5632"}],"number":351180}
buildkite-pr-comment-->
